gitclone/checkout_helper.go

```diff
@@ -72,7 +73,7 @@ func fetch(gitCmd git.Git, remote string, ref string, options fetchOptions) erro
 		return handleCheckoutError(
 			listBranches(gitCmd),
 			fetchFailedTag,
-			fmt.Errorf("fetch failed: %v", err),
```

Comment:
Removing `fetch failed: ` from here since the caller of this method has more context on what went wrong we add additional error messaging at the various call sites instead.